resources/js/chat/conversation-view.tsx

```diff
@@ -66,13 +65,12 @@ export default class ConversationView extends React.Component<Props> {
       }
 
       // check whether the day-change header needs to be shown
-      if (isEmpty(conversationStack) || moment(message.timestamp).date() !== currentDay /* TODO: make check less dodgy */) {
+      if (isEmpty(conversationStack) || moment().isSame(message.timestamp, 'day')) {
```

Comment:
This is supposed to compare with the timestamp of the previous message (store `moment(message.timestamp)` instead of `.date()` in `currentDay` and use `currentDay.isSame`), not the current point in time

or just remove `moment`:
```
new Date(message.timestamp).toLocaleDateString() !== currentDay
.
.
currentDay = new Date(message.timestamp).toLocaleDateString();
```

Reply:
I also forgot a logical not lol

Reply:
Should be fixed, now I'm comparing the good stuff and I'm also added a `!` when using `isSame()`, as we want `true` when we got 2 different days.

Reply:
I forgot to check if currentDay isn't defined bruh, and in this situation the output is incorrect...

Reply:
Finally I did what you suggested, the thing I was doing ignored the timezone, which shouldn't be ignored in this context.